### PR TITLE
Add GPS Expression For UCR

### PR DESCRIPTION
**Pull request by sheelio**
For ICDS reports, needed something to split up the GPS value from CommCare into lat/long.  

Waiting for tests to run, but also not clear if this is the best way to implement this.

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 12 errors:

* corehq/apps/userreports/expressions/factory.py, line 224 - W293 blank line contains whitespace
* corehq/apps/userreports/expressions/factory.py, line 225 - E302 expected 2 blank lines, found 1
* corehq/apps/userreports/expressions/specs.py, line 335 - E302 expected 2 blank lines, found 1
* corehq/apps/userreports/expressions/specs.py, line 341 - W293 blank line contains whitespace
* corehq/apps/userreports/expressions/specs.py, line 349 - F821 undefined name '_'
* corehq/apps/userreports/expressions/specs.py, line 355 - E111 indentation is not a multiple of four
* corehq/apps/userreports/expressions/specs.py, line 356 - E111 indentation is not a multiple of four
* corehq/apps/userreports/expressions/specs.py, line 357 - W293 blank line contains whitespace
* corehq/apps/userreports/expressions/specs.py, line 366 - W293 blank line contains whitespace
* corehq/apps/userreports/expressions/specs.py, line 369 - W293 blank line contains whitespace
* corehq/apps/userreports/tests/test_expressions.py, line 1039 - W293 blank line contains whitespace
* corehq/apps/userreports/tests/test_expressions.py, line 1040 - E302 expected 2 blank lines, found 1

**Comment by czue** [comment]
@sheelio let me know if you want any feedback on this

**Comment by sheelio** [comment]
Fixed issues and re-opening tests.

@czue happy to make changes based on your feedback

File: corehq/apps/userreports/expressions/specs.py [hunk 1]
```diff
@@ -331,3 +331,45 @@ def _get_subcases(self, case_id, context):
...
+    SUPPORTED_GPS_VALUES = [LATITUDE, LONGITUDE, ELEVATION, ACCURACY]
+
+    type = TypeProperty('gps_get_value')
+    gps_expression = DefaultProperty(required=True)
+    gps_type = StringProperty(required=True)
+
+    def configure(self, gps_expression):
+        self._gps_expression = gps_expression
+        if self.gps_type not in GpsValueExpressionSpec.SUPPORTED_GPS_VALUES:
+            raise BadSpecError(_("gps_type'{}' is not valid. Valid options are: {} ").format(
```
Comment (dimagimon) [inline corehq/apps/userreports/expressions/specs.py]: F821 undefined name '_'

Comment (sheelio) [inline corehq/apps/userreports/expressions/specs.py]: I used the same formatting as this code:
https://github.com/dimagi/commcare-hq/blob/477de9759cd140bee99293d0cc996102a53e2941/corehq/apps/userreports/expressions/list_specs.py#L64

File: corehq/apps/userreports/expressions/specs.py [hunk 2]
```diff
@@ -331,3 +331,45 @@ def _get_subcases(self, case_id, context):
...
+                self.gps_type,
+                GpsValueExpressionSpec.SUPPORTED_GPS_VALUES
+            ))
+
+    def __call__(self, item, context=None):
+        gps_value = self._gps_expression(item, context)
+        return self._get_gps_element(gps_value, self.gps_type)
+
+    def _get_gps_element(self, gps_value, gps_type):
```
Comment (dimagimon) [inline corehq/apps/userreports/expressions/specs.py]: F821 undefined name '_'

File: corehq/apps/userreports/expressions/specs.py [hunk 3]
```diff
@@ -331,3 +331,21 @@ def _get_subcases(self, case_id, context):
...
+
+class GpsValueExpressionSpec(JsonObject):
+    type = TypeProperty('gps_get_value')
+    gps_expression = DefaultProperty(required=True)
+    gps_type = StringProperty(required=True)
+
+    def configure(self, gps_expression):
+        self._gps_expression = gps_expression
+        if self.gps_type not in SUPPORTED_GPS_VALUES:
+            raise BadSpecError(_("gps_type '{}' is not valid. Valid options are: {} ").format(
```
Comment (dimagimon) [inline corehq/apps/userreports/expressions/specs.py]: F821 undefined name '_'

**Comment by sheelio** [comment]
@czue 

I guess the more generic way to do this would be to have a "split" expression that takes the index as an argument.  Not sure which is preferable.

**Comment by czue** [comment]
was actually chatting with @kaapstorm about split recently. no strong feelings. fyi - not going to get to this today so feel free to have others review if time-sensitive

**Comment by sheelio** [comment]
Going to close in favour of a split expression (seems more generally useful).

**Pull request closed without merging**